Adjustments and reworking some layouts (#213)
infrawatch/documentation: doc-Service-Telemetry-Framework/modules/proc_creating-openstack-environment-file-for-multiple-clouds.adoc

// it after the module has been published to ensure existing links are not
// broken.
//
// The `context` attribute enables module reuse. Every module's ID includes
// {context}, which ensures that the module has a unique ID even if it is
// reused multiple times in a guide.
//
// Start the title with a verb, such as Creating or Create. See also
// _Wording of headings_ in _The IBM Style Guide_.
[id="creating-openstack-environment-file-for-multiple-clouds_{context}"]
= Creating the OpenStack environment file for multiple clouds

[role="_abstract"]
To label traffic according to the cloud of origin, you must create a configuration with cloud-specific instance names. Create an `stf-connectors.yaml` file and adjust the values of `CeilometerQdrEventsConfig`, `CeilometerQdrMetricsConfig` and `CollectdAmqpInstances` to match the AMQP address prefix scheme.

[NOTE]
If you enabled container health and API status monitoring, you must also modify the `CollectdSensubilityResultsChannel` parameter. For more information, see xref:monitoring-container-health-and-api-status_assembly-advanced-features[].

.Prerequisites

ifdef::include_when_13[]

> * Retrieve the CA certificate from the {MessageBus} deployed by {ProjectShort}. For more information, see xref:getting-ca-certificate-from-stf-for-overcloud-configuration_assembly-completing-the-stf-configuration[].

Reviewer: ```suggestion
* You have retrieved the CA certificate from the {MessageBus} deployed by {ProjectShort}. For more information, see xref:getting-ca-certificate-from-stf-for-overcloud-configuration_assembly-completing-the-stf-configuration[].
```

endif::include_when_13[]

> * Retrieve the {MessageBus} route address. For more information, see xref:retrieving-the-qdr-route-address_assembly-completing-the-stf-configuration[].

Reviewer: ```suggestion
* You have retrieved the {MessageBus} route address. For more information, see xref:retrieving-the-qdr-route-address_assembly-completing-the-stf-configuration[].
```

> * Created the base configuration for {ProjectShort}. For more information, see xref:creating-the-base-configuration-for-stf_assembly-completing-the-stf-configuration[].

Reviewer: ```suggestion
* You have created the base configuration for {ProjectShort}. For more information, see xref:creating-the-base-configuration-for-stf_assembly-completing-the-stf-configuration[].
```

* You have created a unique domain name environment file. For more information, see xref:setting-a-unique-cloud-domain_assembly-completing-the-stf-configuration[].
* You have created the `enable-stf.yaml` to set environment defaults for {ProjectShort}. For more information, see xref:creating-the-base-configuration-for-stf_assembly-completing-the-stf-configuration[].

.Procedure

. Log in to the {OpenStack} undercloud as the `stack` user.

. Create a configuration file called `stf-connectors.yaml` in the `/home/stack` directory.

> . In the `stf-connectors.yaml` file configure the `MetricsQdrConnectors` address to connect to the {MessageBus} on the overcloud deployment. Configure the `CeilometerQdrEventsConfig`, `CeilometerQdrMetricsConfig`, `CollectdAmqpInstances`, and `CollectdSensubilityResultsChannel` topic values to match the AMQP address that you want for this cloud deployment.

Reviewer: ```suggestion
. In the `stf-connectors.yaml` file, configure the `MetricsQdrConnectors` address to connect to the {MessageBus} on the overcloud deployment. Configure the `CeilometerQdrEventsConfig`, `CeilometerQdrMetricsConfig`, `CollectdAmqpInstances`, and `CollectdSensubilityResultsChannel` topic values to match the AMQP address that you want for this cloud deployment.
```

endif::include_when_13[]

    CeilometerQdrEventsConfig:
        driver: amqp
        topic: cloud1-event   # <2>

    CeilometerQdrMetricsConfig:
        driver: amqp
        topic: cloud1-metering   # <3>

    CollectdAmqpInstances:
        cloud1-notify:        # <4>
            notify: true
            format: JSON
            presettle: false
        cloud1-telemetry:     # <5>
            format: JSON
            presettle: true

ifdef::include_when_16[]
    CollectdSensubilityResultsChannel: collectd/cloud1-notify # <6>
endif::include_when_16[]
----
<1> Replace the `host` parameter with the value of `HOST/PORT` that you retrieved in xref:retrieving-the-qdr-route-address_assembly-completing-the-stf-configuration[].
<2> Define the topic for Ceilometer events. This value is the address format of `anycast/ceilometer/cloud1-event.sample`.
<3> Define the topic for Ceilometer metrics. This value is the address format of `anycast/ceilometer/cloud1-metering.sample`.
<4> Define the topic for collectd events. This value is the format of `collectd/cloud1-notify`.
<5> Define the topic for collectd metrics. This value is the format of `collectd/cloud1-telemetry`.
ifdef::include_when_16[]

> <6> Define the topic for collectd-sensubility events. This should be the exact string format of `collectd/cloud1-notify`

Reviewer: ```suggestion
<6> Define the topic for collectd-sensubility events. Ensure that this value is the exact string format `collectd/cloud1-notify`
```
